src/lib/es2015.promise.d.ts

```diff
@@ -1,3 +1,5 @@
+type Awaited<T> = T extends undefined ? T : T extends PromiseLike<infer U> ? U : T;

```

Comment:
I would propose we move this to `es5.d.ts` so that it sits alongside `PromiseLike`, as it has no dependency on `Promise` directly.

Reply:
What is the need for `T extends undefined`?
Wouldn't it works the same without the first conditional?